src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -75,66 +76,203 @@ QIcon icon_ ## FUNC( Gui::BitmapFactory().pixmap(ICONSTR) ); \
 void ElementView::FUNC(){ \
    Gui::Application::Instance->commandManager().runCommandByName(CMDSTR);}
 
-// helper class to store additional information about the listWidget entry.
-class ElementItem : public QListWidgetItem
-{
+
+class ElementWidgetIcons {
+
 public:
-    ElementItem(const QIcon & icon, const QString & text, int elementnr,
-                int startingVertex, int midVertex, int endVertex,
-                Base::Type geometryType, bool construction, bool external)
-        : QListWidgetItem(icon,text)
-        , ElementNbr(elementnr)
-        , StartingVertex(startingVertex)
-        , MidVertex(midVertex)
-        , EndVertex(endVertex)
-        , isLineSelected(false)
-        , isStartingPointSelected(false)
-        , isEndPointSelected(false)
-        , isMidPointSelected(false)
-        , GeometryType(geometryType)
-        , isConstruction(construction)
-        , isExternal(external)
+    enum class Type {
+        Normal,
+        Construction,
+        External,
+        Internal
+    };
+
+private:
+    ElementWidgetIcons()
     {
+        initIcons();
+    }
 
+public:
+    ElementWidgetIcons(const ElementWidgetIcons &) = delete;
+    ElementWidgetIcons(ElementWidgetIcons &&) = delete;
+    ElementWidgetIcons & operator=(const ElementWidgetIcons &) = delete;
+    ElementWidgetIcons & operator=(ElementWidgetIcons &&) = delete;

```

Comment:
It would be more concise to use a user-defined destructor. It is something I have seen recommended by some people.
The intention, however, is better conveyed by making explicit that the object cannot be copied, moved or assigned. I have seen this way recommended by other people.

The outcome is the same, because there are rules defining which constructors and assignment operators a compiler will eventually implicitly create. 

I think that which one to use depends on the need to show intention. This class behaves in a special way and I think this explicit declarations helps understand it is not regular behaviour. Still I would not are to say any of them is wrong.

Reply:
I have no strong opinion, but a user-defined dtor with a comment about its purpose would be more concise. :)
That's of no importance for this PR though.